nise/generators/oci/oci_generator.py

```diff
@@ -90,64 +93,28 @@
 class OCIGenerator(AbstractGenerator):
     """Defines a abstract class for generators."""
 
-    OCI_REGIONS_TO_DOMAIN = [
-        {"region": "ap-sydney-1", "domain": 1},
-        {"region": "ap-melbourne-1", "domain": 1},
-        {"region": "sa-saopaulo-1", "domain": 1},
-        {"region": "sa-vinhedo-1", "domain": 1},
-        {"region": "ca-montreal-1", "domain": 1},
-        {"region": "ca-toronto-1", "domain": 1},
-        {"region": "sa-santiago-1", "domain": 1},
-        {"region": "eu-marseille-1", "domain": 1},
-        {"region": "eu-frankfurt-1", "domain": 3},
-        {"region": "ap-hyderabad-1", "domain": 1},
-        {"region": "ap-mumbai-1", "domain": 1},
-        {"region": "il-jerusalem-1", "domain": 1},
-        {"region": "eu-milan-1", "domain": 1},
-        {"region": "ap-osaka-1", "domain": 1},
-        {"region": "ap-tokyo-1", "domain": 1},
-        {"region": "eu-amsterdam-1", "domain": 1},
-        {"region": "me-jeddah-1", "domain": 1},
-        {"region": "ap-singapore-1", "domain": 1},
-        {"region": "af-johannesburg-1", "domain": 1},
-        {"region": "ap-seoul-1", "domain": 1},
-        {"region": "ap-chuncheon-1", "domain": 1},
-        {"region": "eu-stockholm-1", "domain": 1},
-        {"region": "eu-zurich-1", "domain": 1},
-        {"region": "me-abudhabi-1", "domain": 1},
-        {"region": "me-dubai-1", "domain": 1},
-        {"region": "uk-london-1", "domain": 1},
-        {"region": "uk-cardiff-1", "domain": 1},
-        {"region": "us-ashburn-1", "domain": 3},
-        {"region": "us-phoenix-1", "domain": 3},
-        {"region": "us-sanjose-1", "domain": 1},
-    ]
-
     def __init__(self, start_date, end_date, currency, attributes=None):
         """Initialize the generator."""
         super().__init__(start_date, end_date)
+        if attributes is None or not isinstance(attributes, dict):
+            attributes = {}
         self.currency = currency
```

Comment:
Sorry for all the meandering. My understanding is evolving as I understand the relationship between these objects better.

Having an instance of `OCIReportConstantColumns` attached to an `OCIGenerator` seems to be the best relationship and lines up with how data classes are meant to be used.

```suggestion
        self.currency = currency
        self.constants = OCIReportConstantColumns()
```

Once there is an instance of the constants, things becomes clearer.